flashinfer/cute_dsl/blockscaled_contiguous_grouped_gemm.py

```
@@ -0,0 +1,493 @@
+        # Add group index for each tile in this group
+        tile_idx_to_group_idx_list.extend([i] * num_tiles_in_group)
+
+    num_non_exiting_tiles = len(tile_idx_to_group_idx_list)
+
+    # Apply padding if requested (for CUDA graph support)
+    if permuted_m is not None:
+        if permuted_m < valid_m:
+            raise ValueError(
+                f"permuted_m ({permuted_m}) must be >= valid_m ({valid_m}). "
+                f"Cannot pad to a smaller size."
+            )
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)